### PR TITLE
add Via support and improve default keymap

### PR DESCRIPTION
**Pull request by tominabox1**
## Description

Add VIA support for Le Chiffre and improve the default keymap (similar to QAZ and _33)

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/le_chiffre/keymaps/via/keymap.c [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    // Host Keyboard Layer Status
+    oled_write_P(PSTR("Layer"), false);
+    switch (get_highest_layer(layer_state)) {
+        case _BASE:
+            oled_write_ln_P(PSTR("BASE"), false);
+            break;
+        case _NUM_SYM:
+            oled_write_ln_P(PSTR(" NUM"), false);
+            break;
+        case _NAV:
```
Comment (ridingqwerty) [inline keyboards/le_chiffre/keymaps/via/keymap.c:82]: You're referencing the layer IDs enumerated in the default keymap here, but they don't exist in this keymap.

**Review by ridingqwerty — CHANGES_REQUESTED** [review]
Spurious reference to layer IDs in the OLED rendering of the VIA keymap.

File: keyboards/le_chiffre/keymaps/default/config.h [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#define IGNORE_MOD_TAP_INTERRUPT //helps with homerow mods
```
Comment (drashna) [inline keyboards/le_chiffre/keymaps/default/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

Also, 
```suggestion
#pragma once

#define IGNORE_MOD_TAP_INTERRUPT //helps with homerow mods
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/le_chiffre/keymaps/default/config.h (hunk 2 above)
Comment (tominabox1) [reply keyboards/le_chiffre/keymaps/default/config.h:18]: Thanks for the review

**Review by drashna — APPROVED** [review]

**Comment by tominabox1** [comment]
Please put this on hold. I've updated the design to include on-board RGB so I'd like to add rgb matrix to this before we push forward. Thanks for the review so far.

**Comment by ridingqwerty** [comment]
Placed on hold per @tominabox1 ’s request

**Comment by tominabox1** [comment]
OK We're good to go, Protos worked great!

File: keyboards/le_chiffre/rules.mk [hunk 3]
```diff
@@ -16,7 +16,8 @@ MOUSEKEY_ENABLE = yes       # Mouse keys
 EXTRAKEY_ENABLE = yes       # Audio control and System control
 SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
 NKRO_ENABLE = yes            # USB Nkey Rollover
-RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
 TAP_DANCE_ENABLE = no
```
Comment (fauxpark) [inline keyboards/le_chiffre/rules.mk:19]:
```suggestion
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
TAP_DANCE_ENABLE = no
```

File: keyboards/le_chiffre/rules.mk [hunk 4]
```diff
@@ -16,7 +16,8 @@ MOUSEKEY_ENABLE = yes       # Mouse keys
...
 ENCODER_ENABLE = yes
 OLED_DRIVER_ENABLE = no
+
+RGB_MATRIX_ENABLE = WS2812
```
Comment (fauxpark) [inline keyboards/le_chiffre/rules.mk:23]:
```suggestion
RGB_MATRIX_ENABLE = yes
RGB_MATRIX_DRIVER = WS2812
```

File: keyboards/le_chiffre/keymaps/default/keymap.c [hunk 5]
```diff
@@ -42,101 +65,138 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
...
+    oled_write_ln_P(PSTR("CTRL"), (modifiers & MOD_MASK_CTRL));
+    oled_write_ln_P(PSTR("GUI"), (modifiers & MOD_MASK_GUI));
+}
+
+void oled_task_user(void) {
+    render_lechiffre_logo();
+    oled_set_cursor(0,3);
+    render_layer_status();	// Renders the current keyboard state (layer, lock, caps, scroll, etc)
+	render_mod_status(get_mods()|get_oneshot_mods());
+	render_keylock_status(host_keyboard_leds());
```
Comment (fauxpark) [inline keyboards/le_chiffre/keymaps/default/keymap.c:192]:
```suggestion
	render_keylock_status(host_keyboard_led_state());
```

File: keyboards/le_chiffre/keymaps/default/keymap.c [hunk 6]
```diff
@@ -42,101 +65,138 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
...
+    oled_write(keylog_str, false);
+}
+
+void render_keylock_status(uint8_t led_usb_state) {
+    oled_write_P(PSTR("-----"), false);
+    oled_write_P(PSTR("C"), led_usb_state & (1 << USB_LED_CAPS_LOCK));
+	oled_write_P(PSTR(" "), false);
+    oled_write_P(PSTR("N"), led_usb_state & (1 << USB_LED_NUM_LOCK));
+	oled_write_P(PSTR(" "), false);
+    oled_write_P(PSTR("S"), led_usb_state & (1 << USB_LED_SCROLL_LOCK));
```
Comment (fauxpark) [inline keyboards/le_chiffre/keymaps/default/keymap.c:175]:
```suggestion
void render_keylock_status(led_t led_state) {
    oled_write_P(PSTR("-----"), false);
    oled_write_P(PSTR("C"), led_state.caps_lock);
	oled_write_P(PSTR(" "), false);
    oled_write_P(PSTR("N"), led_state.num_lock);
	oled_write_P(PSTR(" "), false);
    oled_write_P(PSTR("S"), led_state.scroll_lock);
```

File: keyboards/le_chiffre/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+    oled_write(keylog_str, false);
+}
+
+void render_keylock_status(uint8_t led_usb_state) {
+    oled_write_P(PSTR("-----"), false);
+    oled_write_P(PSTR("C"), led_usb_state & (1 << USB_LED_CAPS_LOCK));
+	oled_write_P(PSTR(" "), false);
+    oled_write_P(PSTR("N"), led_usb_state & (1 << USB_LED_NUM_LOCK));
+	oled_write_P(PSTR(" "), false);
+    oled_write_P(PSTR("S"), led_usb_state & (1 << USB_LED_SCROLL_LOCK));
```
Comment (fauxpark) [inline keyboards/le_chiffre/keymaps/via/keymap.c:134]:
```suggestion
void render_keylock_status(led_t led_state) {
    oled_write_P(PSTR("-----"), false);
    oled_write_P(PSTR("C"), led_state.caps_lock);
	oled_write_P(PSTR(" "), false);
    oled_write_P(PSTR("N"), led_state.num_lock);
	oled_write_P(PSTR(" "), false);
    oled_write_P(PSTR("S"), led_state.scroll_lock);
```

File: keyboards/le_chiffre/keymaps/via/keymap.c [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+    oled_write_ln_P(PSTR("CTRL"), (modifiers & MOD_MASK_CTRL));
+    oled_write_ln_P(PSTR("GUI"), (modifiers & MOD_MASK_GUI));
+}
+
+void oled_task_user(void) {
+    render_lechiffre_logo();
+    oled_set_cursor(0,3);
+    // render_layer_status();	// Renders the current keyboard state (layer, lock, caps, scroll, etc)
+	render_mod_status(get_mods()|get_oneshot_mods());
+	render_keylock_status(host_keyboard_leds());
```
Comment (fauxpark) [inline keyboards/le_chiffre/keymaps/via/keymap.c:151]:
```suggestion
	render_keylock_status(host_keyboard_led_state());
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/le_chiffre/rules.mk [hunk 9]
```diff
@@ -11,12 +11,15 @@ MCU = atmega32u4
...
 
-BOOTMAGIC_ENABLE = lite       # Virtual DIP switch configuration
+BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
 MOUSEKEY_ENABLE = yes       # Mouse keys
 EXTRAKEY_ENABLE = yes       # Audio control and System control
 SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
-NKRO_ENABLE = yes            # USB Nkey Rollover
-RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+NKRO_ENABLE = yes           # USB Nkey Rollover
+RGBLIGHT_ENABLE = no 		# Enable keyboard RGB underglow
```
Comment (fauxpark) [inline keyboards/le_chiffre/rules.mk:19]:
```suggestion
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
```

File: keyboards/le_chiffre/keymaps/via/rules.mk [hunk 10]
```diff
@@ -0,0 +1,3 @@
+VIA_ENABLE = yes
+COMBO_ENABLE = no
+OLED_ENABLE = no
```
Comment (fauxpark) [inline keyboards/le_chiffre/keymaps/via/rules.mk:3]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

File: keyboards/le_chiffre/config.h [hunk 11]
```diff
@@ -42,5 +57,24 @@
...
 #define ENCODERS_PAD_B { D3 }
 
-/* #define RGB_DI_PIN F0
-   #define RGBLED_NUM 1  */
+/*
+ * WS2812 Underglow Matrix options
+ */
+#define RGB_DI_PIN F0
+#define RGBLED_NUM 11
+#define RGBLIGHT_LED_MAP { 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 }
```
Comment (zvecr) [inline keyboards/le_chiffre/config.h:65]: I might be missing something, but wouldn't this produce the same as it not being set?
```suggestion
```

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**